libs/designsystem/src/lib/components/list/list.component.ts

```diff
@@ -243,12 +243,15 @@ export class ListComponent implements OnInit, AfterViewInit, OnChanges {
   }
 
   _getBoundaryClass(index: number, section?: any[]): BoundaryClass {
-    const _items = section || this.items;
+    if (index === 0) {
+      return section.length === 1 ? 'first last' : 'first';
+    }
 
-    if (index === 0 || _items[index - 1]?.headingName) return this.headerTemplate ? null : 'first';
```

Comment:
I suspect that the issues with rounded corners, when there is a header/footer is caused, because the checks for `headinName` and `footerTemplate` were removed :)

Reply:
Very good spot on the bug I have refactored so it solves this bug now 🗡️